Cargo.toml

```diff
@@ -204,6 +204,7 @@ unicode-width.workspace = true
 url.workspace = true
 walkdir.workspace = true
 supports-unicode = "3.0.0"
+md-5 = "0.10.6"
```

Comment:
Does this get used anywhere? Only sha256 is mentioned in these commits (there probably isn't much reason to support md5 anyway)

Reply:
I'm not happy about supporting md5 myself, but it is something that my rustc PR could feasibly emit. I'm happy to remove it from both PRs if we can build a consensus on that.

Reply:
This deserves more explanation. Md5 is used by rustc for debug info files, so the support for it was already present in rustc.

Reply:
Makes sense, just doesn’t look like it is used anywhere in this diff?

Reply:
It's imported in src/cargo/core/compiler/fingerprint/mod.rs, which github hides by default due to the size of the diff.

Reply:
It gets used here [Line 380 of src/cargo/core/compiler/fingerprint/mod.rs](https://github.com/rust-lang/cargo/pull/14137/files#diff-33fc5e62ca2fa6b54fd70249825e4f28dcc490317cd3a89e3c07aa62754406ceR380)

Reply:
The only reason that md5 and sha-1 were included as options for the debuginfo checksum was compatibility with PDB (pre-sha256 support) & DWARF5 debug formats.

Since we're making something new here, I don't think there's any reason to support deprecated hashing algorithms.

Reply:
> It gets used here [Line 380 of src/cargo/core/compiler/fingerprint/mod.rs](https://github.com/rust-lang/cargo/pull/14137/files#diff-33fc5e62ca2fa6b54fd70249825e4f28dcc490317cd3a89e3c07aa62754406ceR380)

🤦‍♂️ didn't see the collapsed file when I ctrl+F'd for it

Reply:
md5 and sha1 are no longer supported for this. I've also added an option to use xxhash. xxhash is not cryptographically secure, but that's not necessary for this use case. So I figured it was worth comparing the performance of it to sha256.

Reply:
xxhash has been swapped out for blake3. It was decided during the rustc MCP that cryptographically secure hashes are necessary in order for this freshness algorithm to be correct.

Reply:
Blake3 appears to more mature over-all anyway as it has runtime CPU feature detection for SSE2/SSE4.1/AVX2/AVX-512. The most commonly used XXH library only supports compile-time configuration of these which makes it less suited to a tool like cargo that's compiled once and ran on a wide variety of hardware.

When I was benchmarking algorithms for a similar purpose (source fingerprinting in go-task) both Blake3 and XXH were fast enough that the bottleneck was almost always disk IO and the runtime differences were minimal.